src/services/AdvancedTradingView/advancedTradingView.selectors.ts

```diff
@@ -0,0 +1,259 @@
+import { Pair, RoutablePlatform, Token, UniswapV2RoutablePlatform } from '@swapr/sdk'
+
+import { createSelector } from '@reduxjs/toolkit'
+
+import { AppState } from '../../state'
+import { BaseAppState } from './adapters/baseAdapter/base.adapter'
+import { AllTradesAndLiquidityFromAdapters, BasePair, LiquidityTypename } from './adapters/baseAdapter/base.types'
+import { UniswapV3Pair } from './adapters/uniswapV3/uniswapV3.types'
+import { AdapterKeys, AdvancedViewTransaction } from './advancedTradingView.types'
+
+const adapterLogos: { [key in AdapterKeys]: string } = {
+  swapr: UniswapV2RoutablePlatform.SWAPR.name,
+  sushiswap: UniswapV2RoutablePlatform.SUSHISWAP.name,
+  uniswapV2: UniswapV2RoutablePlatform.UNISWAP.name,
+  honeyswap: UniswapV2RoutablePlatform.HONEYSWAP.name,
+  uniswapV3: RoutablePlatform.UNISWAP.name,
+}
+
+const formatAdapterAmount = (amount: Number) => {
+  if (amount < 0.000001) return amount.toFixed(4)
+
+  if (amount < 0.001) return amount.toFixed(6)
+
+  return amount.toFixed(4)
+}
+
+export const sortsBeforeTokens = (inputToken: Token, outputToken: Token) =>
+  inputToken.sortsBefore(outputToken) ? [inputToken, outputToken] : [outputToken, inputToken]
+
+export const getAdapterPair = <T extends BaseAppState>(key: AdapterKeys, platform: UniswapV2RoutablePlatform) =>
+  createSelector(
+    [(state: T) => state.advancedTradingView.pair, (state: T) => state.advancedTradingView.adapters[key]],
+    ({ inputToken, outputToken }, adapterPairs) => {
+      if (inputToken && outputToken) {
+        try {
+          const pairId = Pair.getAddress(inputToken, outputToken, platform).toLowerCase()
+          return {
+            pair: adapterPairs[pairId] as BasePair,
+            logoKey: adapterLogos[key],
+          }
+        } catch {}
+      }
+    }
+  )
+const getAdapterPairUniswapV3 = (key: AdapterKeys) =>
+  createSelector(
+    [(state: AppState) => state.advancedTradingView.pair, (state: AppState) => state.advancedTradingView.adapters[key]],
```

Comment:
Just to remember. Better to use the same typing as above.

Reply:
fixed [501266b](https://github.com/levelkdev/swapr-dapp/pull/1503/commits/501266b3e2ac4a76be7811d2fa67025ee4d1d58b)